giraph-core/src/main/java/org/apache/giraph/bsp/BspService.java

```diff
@@ -425,11 +431,20 @@ public final String getWorkerWroteCheckpointPath(long attempt,
    *
    * @param attempt application attempt number
    * @param superstep superstep to use
+   * @param readWriteCounters Boolean to denote whether we are
+   *                          reading/writing counters
    * @return directory path based on the a superstep
    */
-  public final String getWorkerFinishedPath(long attempt, long superstep) {
-    return applicationAttemptsPath + "/" + attempt +
+  public final String getWorkerFinishedPath(long attempt, long superstep,
```

Comment:
Let's make a separate method that just returns the counters path. It seems clearer to separate these two.